gammapy/image/core.py

```diff
@@ -298,6 +298,49 @@ def contains(self, position):
         x, y = skycoord_to_pixel(position, self.wcs, self.wcs_origin)
         return (x >= 0.5) & (x <= nx + 0.5) & (y >= 0.5) & (y <= ny + 0.5)
 
+    def footprint(self, mode='center'):
+        """
+        Calculates the footprint of the image on the sky.
+
+        Parameters
+        ----------
+        mode : {'center', 'edge'} (default 'center')
+            Return the positions of the corners of the image on the sky by
+            using the center of the pixel or the edge.
+
+        Returns
+        -------
+        coordinates : `~collections.OrderedDict`
+            Dictionary of the positions of the corners of the image
+            with keys {'lower left', 'lower right', 'upper left', 'upper right'}
+            and `~astropy.coordinates.SkyCoord` objects as values.
+
+        Examples
+        --------
+        >>> from gammapy.image import SkyImage
+        >>> image = SkyImage.empty(nxpix=3, nypix=2)
+        >>> coord = image.footprint(mode='center')
+        >>> coord['lower left']
+        <SkyCoord (Galactic): (l, b) in deg
+            (359.96, -0.01)>
+        """
+        naxis2, naxis1 = self.data.shape
+        if mode == 'center':
+            pixcoord = [(0, 0), (naxis1, 0), (0, naxis2), (naxis1, naxis2)]
+        elif mode == 'edge':
+            pixcoord = [(-0.5, -0.5), (naxis1 + 0.5, -0.5),
+                         (-0.5, naxis2 + 0.5), (naxis1 + 0.5, naxis2 + 0.5)]
+        else:
+            raise ValueError('Invalid mode: {}'.format(mode))
+
+        coordinates = OrderedDict()
+        keys = ['upper left', 'lower left', 'upper right', 'lower right']
```

Comment:
I think, the order is correct, but it will be better if you have a look on it. 
`SkyCoord.from_pixel()` just convert pixel to `SkyCoord` object and doesn't have any information about the order of pixels. I construct this order on base of results of `calc_footprint` method.